ql/extractor/Cargo.toml

```diff
@@ -2,19 +2,22 @@
 name = "codeql-extractor-ql"
 version = "0.1.0"
 authors = ["GitHub"]
-edition = "2018"
+edition = "2021"
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-tree-sitter = ">= 0.20, < 0.21"
-tree-sitter-ql = { git = "https://github.com/tree-sitter/tree-sitter-ql.git", rev = "d08db734f8dc52f6bc04db53a966603122bc6985"}
-tree-sitter-ql-dbscheme = { git = "https://github.com/erik-krogh/tree-sitter-ql-dbscheme.git", rev = "63e1344353f63931e88bfbc2faa2e78e1421b213"}
+tree-sitter = ">= 0.22.6"
+tree-sitter-ql = { git = "https://github.com/tree-sitter/tree-sitter-ql.git", rev = "fa5c3821dd2161f5c8528a8cbdb258daa6dc4de6"}
+tree-sitter-ql-dbscheme = { git = "https://github.com/tree-sitter/tree-sitter-ql-dbscheme.git", rev = "5f770f57fa415607ff50e3d237d47c8f11440eb3"}
 tree-sitter-blame = {path = "../buramu/tree-sitter-blame"}
-tree-sitter-json = {git = "https://github.com/tausbn/tree-sitter-json.git", rev = "745663ee997f1576fe1e7187e6347e0db36ec7a9"}
+tree-sitter-json = {git = "https://github.com/tree-sitter/tree-sitter-json.git", rev = "94f5c527b2965465956c2000ed6134dd24daf2a7"}
```

Comment:
Note that this will break structured log extraction. Some of the commits on my fork added support for things like having multiple JSON objects/arrays in the same file (i.e. JSONC/JSONL), and some added support for trailing commas in objects/arrays. Both of these are still missing on mainline `tree-sitter-json`, as far as I know.

This is not to say that this PR should be blocked, however. The structured log parsing is not in active use (and to be honest, it should be replaced with a better solution than (ab)using the tree-sitter interface). I just wanted to point out that this _is_ a breaking change.

Reply:
Thanks. Perhaps worth submitting an upstream PR to `tree-sitter/tree-sitter-json`?